### DISCUSSION TITLE
[Enhancement] Add event link to /dev/input/by-id

### DISCUSSION THREAD

**Discussion by LunaticWasTaken** [xremap/xremap#634] [Ideas]
Currently, there is only an event file being created (eg. "event26"), but if i see this correctly, this event name is never the same (eg. when hotplugging devices).

In my specific case, i need to pass through my keyboard (or in this case the xremap device) to one of my VM's, and i have absolutely 0 way of identifying the event that file used by xremap.

Adding such a link at xremap-startup (and removing it on xremap-shutdown) would be helpful in such cases, since it would make it possible to identify the device

**Comment by k0kubun** [comment]
> this event name is never the same (eg. when hotplugging devices).

I don't get this part. If you're just unplugging/plugging a device, isn't the name going to be the same? If so, using `--watch` should be enough.

**Comment by LunaticWasTaken** [comment]
> > this event name is never the same (eg. when hotplugging devices).
> 
> I don't get this part. If you're just unplugging/plugging a device, isn't the name going to be the same?

This seems to be the case for stuff like my Mouse, but in the case of XRemap, it seems that it always becomes `event<event-amount + 1>`. This might be event26, might also be event27, entirely depending on how many devices are plugged in when starting xremap.

Also i don't get how --watch would help. What i need is a static identifier for the input/uinput/whatever device, eg:
```
    <input type='evdev'>
      <source dev='/dev/input/by-id/KEYBOARD_NAME' grab='all' repeat='on' grabToggle='ctrl-ctrl'/>
    </input>
```
where `KEYBOARD_NAME` has to be a static name. It is entered into the libvirt config exactly as that, and has to match exactly.
Normally i'd just use `/dev/input/by-id/usb-Razer_Razer_BlackWidow_V3_Tenkeyless-event-kbd` for my keyboard, but since that is grabbed by xremap, it does not give any input events anymore.
